chainpy/eth/managers/contracthandler.py

```diff
@@ -43,56 +43,74 @@ def __init__(
         )
 
         self._matured_latest_height = DEFAULT_LATEST_HEIGHT if latest_height is None else latest_height
-        self.__max_log_num = DEFAULT_MAX_LOG_NUM if max_log_num is None else max_log_num
+        self._max_log_num = DEFAULT_MAX_LOG_NUM if max_log_num is None else max_log_num
 
         self._contracts = dict()
-        self._contract_name_by_event_name = dict()
-        self._event_name_by_topic = dict()
+        self._event_db: Dict[str, List[Dict[str, Any]]] = dict()
 
         for contract_dict in contracts:
+            # determine abi_path of the contract
             abi_path = contract_dict.get("abi_path")
             if abi_path is None:
                 abi_path = abi_dir + contract_dict.get("abi_file")
+
+            # init contract_obj of the contract
             contract_obj = EthContract.from_abi_file(contract_dict["name"], contract_dict["address"], abi_path)
-            self._contracts[contract_dict["name"]] = contract_obj
+
+            # assume that every contract has a different name.
+            contract_name = contract_dict["name"]
+            if self._contracts.get(contract_name) is not None:
+                raise Exception("Collision contract names")
+
+            # store contract obj by contract_name
+            self._contracts[contract_name] = contract_obj
 
         if events is not None and isinstance(events, list):
             for event in events:
-                # parse contract's address and abi
-                event_name = event["event_name"]
-                self._contract_name_by_event_name[event_name] = event["contract_name"]
-
-                contract = self._contracts[event["contract_name"]]
-                topic = contract.get_method_abi(event_name).get_topic()
-                self._event_name_by_topic[topic.hex()] = event_name
+                # Check that each event belongs to a known contract.
+                contract_name = event["contract_name"]
+                if self._contracts.get(contract_name) is None:
+                    raise Exception("Event of unknown contract: {}".format(event))
+
+                event_name, contract_name = event["event_name"], event["contract_name"]
```

Comment:
contract_name l.71에서 정의 했는데 또 하네요
```suggestion
                event_name, contract_name = event["event_name"], event["contract_name"]
                if self._contracts.get(contract_name) is None:
                    raise Exception("Event of unknown contract: {}".format(event))
```